api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -338,6 +338,23 @@ message Principal {
     type.matcher.v3.StringMatcher principal_name = 2;
   }
 
+  message FilterStateIPMatcherConfig {
+    // The key at which the filter state object is stored.
+    string key = 1;
+
+    core.v3.CidrRange ip_range = 2;
+  }
+
+  message DynamicMetadataIPMatcherConfig {
```

Comment:
This seems like a fairly specific use case for the general mechanism provided by the existing `SourcedMetadata` matcher.  Instead of adding this special case, can we generalize `type.matcher.v3.MetadataMatcher` to allow arbitrary plugin matchers, and then define a plugin that matches an IP range?